### DISCUSSION TITLE
Is it possible to save the flow as an svg file?

### DISCUSSION THREAD

**Discussion by ghost** [xyflow/xyflow#1139] [Q&A] [answered]
I would like to know if there is a way to export the flow as svg?

**Comment by moklick** [comment]
It depends what you need. The flow itself is a mix of SVG and DOM nodes but the Minimap is SVG only. You could also write your own static renderer for the elements.

**Reply by ghost** [reply]
I need to get the flow itself as an svg file, the minimap would not be enough, i'm fairly new to react and typescript so could you please give me some more insights into how to statically render the elements out? Are the svg parts of the flow accessible in any way?

**Reply by moklick** [reply]
the nodes are DOM elements and the edges are SVG paths, so it's not possible to get the whole graph as a SVG. A static renderer means that you would need to re-implement your nodes as SVGs. At the moment it's very complicated unfortunately. Maybe we will have a SVG renderer in the future.

**Reply by ghost** [reply]
Thanks for the Info, would it be possible to just get the dom Elements and use another library to convert to svg and if so, do you know any convenient solutions?

**Reply by ghost** [reply]
@moklick Hi, I just wanted to ask, if there is something new by any chance, and if not, How can I get a list of the nodes as DOM elements from the flow, so i can try to figure out something on my own?

**Reply by datoslabs** [reply]
@konsti-papa , do you need to save the flow as an image (png or jpeg) or do you need to save the flow as an svg file?  The reason I asked is that it would be relatively easy using html2canvas to save the flow as a png or jpeg file; however, if the output has to be svg, then I am not sure it's easily achievable as you most likely need to find a bitmap trace library to create svg from a png/jpeg image.

BTW, to get a list of DOM elements, you simply need to pass a ref object to the ReactFlow component (see [React.useRef()](https://reactjs.org/docs/hooks-reference.html#useref) for details); the ref object should be the `div.reactflow-wrapper` element, from there, you should be able to grab all the nodes' dom elements via selectors.

**Reply by nadiTime** [reply]
@datoslabs have you figured how to export the graph as a SVG?

**Comment by moklick** [comment, marked as answer]
I've created a ["Download Image" example](https://reactflow.dev/docs/examples/misc/download-image/) that uses [html-to-image](https://github.com/bubkoo/html-to-image) and exports a png but the library also has a SVG export option.

**Comment by 3rd** [comment]
Made it export perfectly for https://tsdiagram.com by using an iframe, if anyone needs this check out:
https://github.com/3rd/tsdiagram/blob/dbd7cdfbe445c2b16a6159b2c822429b4e681639/src/utils/svg-export.ts
